### DISCUSSION TITLE
Laravel 8.16 - Duplicate/Multiple Prepare statements per execute?

### DISCUSSION THREAD

**Discussion by alturic** [laravel/framework#35914] [Q&A]
Running Laravel 8.16, on MySQL, and I'm noticing multiple prepare statements, for every execute, and I'm trying to find out what's causing this.

I'm seeing things like:

```
2021-01-15T21:51:28.857545Z 264 Prepare select * from `users` where `id` = ? limit 1
2021-01-15T21:51:28.857661Z 264 Prepare select * from `users` where `id` = ? limit 1
2021-01-15T21:51:28.857680Z 264 Execute select * from `users` where `id` = 1 limit 1
```

Same with session updates, interestingly the selection of the session data looks to be a single prepare/execute. If I run this:

```
DB::select('SELECT * from users where id = 1');
```

it's a single prepare/execute. However, if I run this:

```
DB::select('SELECT * from users where id = ?', [1]);
```

then it's two prepares and an execute. Using model methods, it's two prepares and an execute...

**Comment by sportinsightsmedia** [comment]
Did you get to the bottom of this? I have just noticed the same thing, using Laravel 8.80.0 and MySQL on AWS Lightsail

**Comment by vrgblzs** [comment]
Same here, Laravel v10.48.11, php v8.2.0, mysql v8.0.31 on a local windows machine

**Comment by vrgblzs** [comment]
I used the built-in mysql general log. On some queries the prepare got duplicated, on some other places it doesn't. 

Example
2024-07-12T09:52:54.373303Z	 3250 Prepare	select * from `languages` where `languages`.`id` = ? limit 1
2024-07-12T09:52:54.373489Z	 3250 Prepare	select * from `languages` where `languages`.`id` = ? limit 1
2024-07-12T09:52:54.373514Z	 3250 Execute	select * from `languages` where `languages`.`id` = 1 limit 1
2024-07-12T09:52:54.373817Z	 3250 Close stmt

**Comment by ghost** [comment]
Go here https://github.com/laravel/framework/blob/11.x/src/Illuminate/Database/Connection.php#L410
and search for ->prepare(

That could be the issue

**Reply by ghost** [reply]
$this->prepared( is called only in 3 places and that is the one that triggers the event.

**Comment by ghost** [comment]
@vrgblzs what pdo extension are you using in your docker?

**Reply by ghost** [reply]
see this maybe it is from the pdo extension https://github.com/laravel/framework/discussions/51844

**Reply by vrgblzs** [reply]
Thanks for the help, it is not a Laravel problem.

It seems, the second prepare is actually a reprepare - if the bound type is not the expected (eg signed and unsigned integer types)
source: https://github.com/php/php-src/issues/11744

**Reply by alturic** [reply]
I wonder if Doctrine doesn't have the issue because you explicitly declare the types in the class?
